_posts/2021-05-19-#18418.md

```diff
@@ -0,0 +1,110 @@
+---
+layout: pr
+date: 2021-05-19
+title: "Increase OUTPUT_GROUP_MAX_ENTRIES to 100"
+pr: 18418
+authors: [fjahr]
+components: ["wallet"]
+host: glozow
+status: upcoming
+commit: 241e141
+---
+
+## Notes
+
+* The way our wallet constructs transactions over time can leak information
+  about its contents.  The most obvious example is we can assume that all UTXOs
+to the same address are controlled by the same person. UTXOs sent to different
+addresses may also be linked if they are spent together (a common heuristic
+used in chain analysis).  Thus, if we're not careful, observant attackers can
+link addresses to estimate our wallet balance and, if any one of our addresses
+is deanonymized (e.g. we send it to an exchange, merchant, or block explorer
+that knows our personal information or IP address), we might accidentally
+reveal how much money we have!
+
+* The Bitcoin Core wallet implements a few best-practice privacy techniques.
+  One is avoiding the reuse of addresses when creating an invoice or change
+address.  Another is grouping UTXOs into
+[OutputGroup](https://github.com/bitcoin-core-review-club/bitcoin/blob/4ac1adda9914d845aaea5804af4801ffec53c701/src/wallet/coinselection.h#L72)s
+by scriptPubKey and spending them together.
+
+* However, `OutputGroup`s can grow quite large - after all, anyone can send as
+  many transactions as they want to an address that we've published. It might
+not make sense to fund a 0.015BTC transaction by sweeping a group of 150 inputs
+worth 10BTC (not to mention the extra fees for all the unnecessary inputs).
+The `OUTPUT_GROUP_MAX_ENTRIES` constant limits the number of UTXOs per
+`OutputGroup`. Within `GroupOutputs()`, if a scriptPubKey exceeds this many, we
```

Comment:
"if a scriptPubKey exceeds this many" -> "if we have more than `OUTPUT_GROUP_MAX_ENTRIES` UTXOs with the same scriptPubKey"